UM-USElab/gradecraft-development
# Caching review

### Status
**IN DEVELOPMENT**

### Description
I added 'touch: true' to the announcement.rb model file. This appears to repair an issue where newly created announcements were not appearing for the user, unless they logged out of GC and then back in.

The specific issue is explained here: 
https://trello.com/c/AAK3C4wY/76-new-announcements-not-being-added-to-the-announcements-table-until-you-leave-the-course-and-return

Original submissions and considerations for this feature involved removing caching from the announcements.rb save pattern completely; however, this PR looks to address the caching issue and restore the functionality.


### Todos


### Deploy Notes
Notes regarding deployment the contained body of work.  These should note any
db migrations, etc.

### Gem dependencies
None

### Migrations
NO

### Steps to Test or Reproduce
Outline the steps to test or reproduce the PR here.

1. In your local development branch run this command to enable caching:
* bundle exec rails dev:cache
* visit the courses in your GC env.
* create a new announcement - check the announcement table for latest announcement.
* without the touch: true being present in the announcement.rb file, the latest announcement will not appear.

### Impacted Areas in Application
List general components of the application that this PR will affect:

* announcements, courses

======================
